kotlinx-coroutines-core/common/src/CoroutineDispatcher.kt

```diff
@@ -149,6 +150,42 @@ public abstract class CoroutineDispatcher :
      * ```
      * Note that there is no guarantee that the underlying system thread will always be the same.
      *
+     * #### It is not a mutex!
+     *
+     * **Pitfall**: [limitedParallelism] limits how many threads can execute some code in parallel,
+     * but does not limit how many coroutines execute concurrently!
+     *
+     * For example:
+     *
+     * ```
+     * val notAMutex = Dispatchers.Default.limitedParallelism(1)
+     *
+     * repeat(3) {
+     *     launch(notAMutex) {
+     *         println("Coroutine $it entering...")
+     *         delay(20.milliseconds)
+     *         println("Coroutine $it leaving.")
+     *     }
+     * }
+     * ```
+     *
+     * The output will be similar to this:
+     *
+     * ```
+     * Coroutine 0 entering...
+     * Coroutine 1 entering...
+     * Coroutine 2 entering...
+     * Coroutine 0 leaving.
+     * Coroutine 1 leaving.
+     * Coroutine 2 leaving.
+     * ```
+     *
+     * This means that coroutines are not guaranteed to run to completion before the dispatcher starts executing
+     * code from another coroutine.
+     * The only guarantee in this example is that two `println` calls in several threads simultaneously.
```

Comment:
Seems to be missing a few words
```suggestion
     * The only guarantee in this example is that two `println` calls will not occur in several threads simultaneously.
```